extensions/standard-processors/tests/unit/EvaluateJsonPathTests.cpp

```diff
@@ -0,0 +1,330 @@
+  REQUIRE(result.at(processors::EvaluateJsonPath::Unmatched).size() == 1);
+  REQUIRE(result.at(processors::EvaluateJsonPath::Failure).empty());
+
+  const auto result_flow_file = result.at(processors::EvaluateJsonPath::Unmatched).at(0);
+
+  CHECK(controller_.plan->getContent(result_flow_file) == "{}");
+
+  std::string attribute_value;
+  CHECK_FALSE(result_flow_file->getAttribute("attribute", attribute_value));
+
+  if (warn_path_not_found_behavior) {
+    CHECK(utils::verifyLogLinePresenceInPollTime(0s, "JSON path '$.firstName' not found for attribute key 'attribute'"));
+  }
+}
+
+TEST_CASE_METHOD(EvaluateJsonPathTestFixture, "JSON path query result does not match the required return type", "[EvaluateJsonPathTests]") {
+  controller_.plan->setDynamicProperty(evaluate_json_path_processor_, "attribute", "$.name");
+
+  SECTION("Return type is set to scalar automatically when destination is set to flowfile-attribute") {
+    controller_.plan->setProperty(evaluate_json_path_processor_, processors::EvaluateJsonPath::Destination, "flowfile-attribute");
+  }
+
+  std::string json_content = R"({"name": {"firstName": "John", "lastName": "Doe"}})";
+  auto result = controller_.trigger({{.content = json_content}});
+
+  REQUIRE(result.at(processors::EvaluateJsonPath::Matched).empty());
+  REQUIRE(result.at(processors::EvaluateJsonPath::Unmatched).empty());
+  REQUIRE(result.at(processors::EvaluateJsonPath::Failure).size() == 1);
+
+  const auto result_flow_file = result.at(processors::EvaluateJsonPath::Failure).at(0);
+
+  CHECK(controller_.plan->getContent(result_flow_file) == json_content);
+  std::string attribute_value;
+  CHECK_FALSE(result_flow_file->getAttribute("attribute", attribute_value));
+  CHECK(utils::verifyLogLinePresenceInPollTime(0s, "JSON path '$.name' returned a non-scalar value or multiple values for attribute key 'attribute', transferring to Failure relationship"));
+}
+
+TEST_CASE_METHOD(EvaluateJsonPathTestFixture, "Query JSON object and write it to flow file", "[EvaluateJsonPathTests]") {
+  controller_.plan->setProperty(evaluate_json_path_processor_, processors::EvaluateJsonPath::Destination, "flowfile-content");
+  controller_.plan->setDynamicProperty(evaluate_json_path_processor_, "jsonPath", "$.name");
+
+  std::string json_content = R"({"name": {"firstName": "John", "lastName": "Doe"}})";
+  auto result = controller_.trigger({{.content = json_content}});
+
+  REQUIRE(result.at(processors::EvaluateJsonPath::Matched).size() == 1);
+  REQUIRE(result.at(processors::EvaluateJsonPath::Unmatched).empty());
+  REQUIRE(result.at(processors::EvaluateJsonPath::Failure).empty());
+
+  const auto result_flow_file = result.at(processors::EvaluateJsonPath::Matched).at(0);
+
+  CHECK(controller_.plan->getContent(result_flow_file) == R"({"firstName":"John","lastName":"Doe"})");
+  std::string attribute_value;
+  CHECK_FALSE(result_flow_file->getAttribute("jsonPath", attribute_value));
```

Comment:
we could use the overload which returns an optional:
```suggestion
  CHECK_FALSE(result_flow_file->getAttribute("jsonPath"));
```